certbot/plugins/selection.py

```diff
@@ -108,11 +108,19 @@ def choose_plugin(prepared, question):
     opts = [plugin_ep.description_with_name +
             (" [Misconfigured]" if plugin_ep.misconfigured else "")
             for plugin_ep in prepared]
+    names = set(plugin_ep.name for plugin_ep in prepared)
 
     while True:
         disp = z_util(interfaces.IDisplay)
-        code, index = disp.menu(
-            question, opts, force_interactive=True)
+        if "CERTBOT_AUTO" in os.environ and names == set(("apache", "nginx")):
```

Comment:
Nope, I missed that. I think perhaps there's an argument for having One Official Way™ to check if we're `certbot-auto` or not, but that should probably be a function not a magic test, and doesn't need to be in this PR.